src/site/antora/modules/ROOT/pages/manual/markers.adoc

```diff
@@ -16,85 +16,111 @@
+
+By using Markers, it is possible to filter log statements based on the Marker
+and display only those log statements that are of interest, such as those
+related to XML processing or SQL queries.
+
+Markers offer more fine-grained control over log filtering beyond log levels or package names.
+
+== Creating Markers
+
+To create a Marker, use the MarkerManager class. The MarkerManager class provides 
+a factory method to create a Marker object is used to tag log statements.
+
+[source, java]
+----
+Marker SQL_MARKER = MarkerManager.getMarker("SQL");
+----
+
+Since a Marker is reusable across multiple log statements, it makes sense 
+to store the Marker in a static final field and make it a constant.
+Once done, use it as the first argument in the log statement.
+
+[source, java]
+----
+logger.debug(SQL_MARKER, "Here is my SQL related message");
+----
+
+A complete example is shown below. It creates the Marker as the constant
+and then uses it to tag a log statement. The log method works "as usual,"
+but the marker needs to be added as the first argument.
+
+[source, java]
 ----
 import org.apache.logging.log4j.Logger;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.MarkerManager;
-import java.util.Map;
 
 public class MyApp {
+    private Logger logger = LogManager.getLogger();
+    private static final Marker SQL_MARKER = MarkerManager.getMarker("MY_APP_SQL"); <1>
+
+    public void doQuery(String table) {
+        // do business logic here        
 
-    private Logger logger = LogManager.getLogger(MyApp.class.getName());
-    private static final Marker SQL_MARKER = MarkerManager.getMarker("SQL");
-    private static final Marker UPDATE_MARKER = MarkerManager.getMarker("SQL_UPDATE").setParents(SQL_MARKER);
-    private static final Marker QUERY_MARKER = MarkerManager.getMarker("SQL_QUERY").setParents(SQL_MARKER);
+        logger.debug(SQL_MARKER, "SELECT * FROM {}", table); <2>
+    }
+}
+----
+<1> The marker is created and stored as a constant
+<2> Using the marker to tag a log statement
 
-    public String doQuery(String table) {
-        logger.traceEntry();
+You must create a configuration file to write log statements tagged with the "SQL" marker to the console. 
+This file uses the `<MarkerFilter>` element to accept or deny log statements 
+based on the marker. 
 
-        logger.debug(QUERY_MARKER, "SELECT * FROM {}", table);
+The following code shows how the MarkerFilter is used in the configuration file. 
+It assumes that the Console appender is defined in the configuration file.
 
-        String result = ... 
+[source, xml]
+----
+<Logger name="com.example" level="all" additivity="false">
+    <AppenderRef ref="Console">
+        <Filters>
+            <MarkerFilter marker="MY_APP_SQL" onMatch="ACCEPT" onMismatch="DENY"/> <1>
+        </Filters>
+    </AppenderRef>
+</Logger>
+----
+<1> The MarkerFilter is used to accept only SQL markers but deny others
```

Comment:
It's very distractive and contains a lot of things that are not relevant to this document.
Either we expand on the explanation (it could be a separate thing or a guide) or we link to a complete example in some other code repository. I am in favor too creating an example repository for code like this.

Reply:
I think that:

* a complete example should be in the `examples` folder,
* the example should be in XML, JSON, YAML and properties (yes, share my pain :wink:),
* only the `Logger` element should be showed in the text,
* I prefer using filters on appender references instead of appenders,
* the composite filter `Filters` plugin is not necessary in this case.

Reply:
I don't know how to _"only show the `Logger` element but also provide a way to obtain the rest of the source code"_ in Antora. Hence, I suggest the following:

```suggestion
Once a log event is tagged with a marker, we can apply specialized configuration on that particular marker.
For instance, we can use a xref:manual/filters.adoc#MarkerFilter[`MarkerFilter`] to redirect log events whose marker matches a certain name to a dedicated appender.
In the example below, all `MY_SQL_APP`-marked log messages from the `com.example` package will be redirected to the `CONSOLE` appender:

[source, xml]
----
<logger name="com.example" level="OFF"><!--1-->
  <AppenderRef ref="CONSOLE">
    <Filters>
      <MarkerFilter marker="MY_APP_SQL" onMatch="ACCEPT" onMismatch="DENY"/><!--2-->
    </Filters>
  </AppenderRef>
</logger>
----
<1> Match all messages from `com.example` package from all levels
<2> Only allow `MY_APP_SQL`-marked messages
```

Note that in the original example `level="all"` was incorrect and XML tags did not comply with our XSD.